lib/package.py

```diff
@@ -139,8 +139,11 @@ def _load(self):
             source_name = source.keys()[0]
             if 'archive' not in source.values()[0]:
                 source[source_name]['archive'] = self.name
-            if 'ref_to_fetch' not in source.values()[0] and source_name == 'git':
-                source[source_name]['ref_to_fetch'] = None
+            if source_name == 'git':
+                if 'ref_to_fetch' not in source.values()[0]:
+                    source[source_name]['ref_to_fetch'] = None
+                if 'archive_src_dir' not in source.values()[0]:
+                    source[source_name]['archive_src_dir'] = None
```

Comment:
The previous comment applies here too, if you accept it. Both nested ifs could be removed.